src/main/java/dev/openfeature/sdk/StatefulFeatureProvider.java

```diff
@@ -0,0 +1,99 @@
+package dev.openfeature.sdk;
+
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+import lombok.experimental.Delegate;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+class StatefulFeatureProvider implements FeatureProvider, EventProviderListener {
```

Comment:
As we discussed, let's remove this implementation of `FeatureProvider`, and the delegation of the associated methods.

Instead, as @guidobrei suggests, let's make this a true wrapper class that adds statefulness and encapsulates a `FeatureProvider` and providers an accessor for it. Then we can avoid having to implement the equality overrides such as [this](https://github.com/open-feature/java-sdk/pull/1096/files#diff-7bbe5cc5c58570709c6e4af7221fde633a9171736c3992c16fe9ff5d78fa75e2R69-R79) and [this](https://github.com/open-feature/java-sdk/pull/1096/files#diff-0bc88e6e0f33e7a2876841d5a59be2ceb032f4600719d8c7668728e473aec94eR70-R79).

Reply:
I `FeatureProviderStateManager` or something like that might be a good name since this will no longer actually be a provider.